### PR TITLE
drivers: spi: spi-dw cleanup and serial target binding

### PR DESCRIPTION
**Pull request by XenuIsWatching**
This cleans up the instantiation macro. This also adds a binding if the IP is configured as a serial target and fixes return of the rx_len when it is a serial target. DBG_COUNTER was also removed as that appears to be unnecessary.

aux-reg should be defined with a hyphen in the bindings

The HAS_SPI_DW Kconfig is rather unnecessary. If the synopsys designware spi is to be included. It should come from the devicetree.

### CODE REVIEW

File: drivers/spi/spi_dw.c [hunk 1]
```diff
@@ -439,12 +421,26 @@ static int transceive(const struct device *dev,
...
 	set_bit_ssienr(info);
 
 	ret = spi_context_wait_for_completion(&spi->ctx);
+
+#ifdef CONFIG_SPI_SLAVE
+	if (spi_context_is_slave(&spi->ctx) && !ret) {
+		ret = spi->ctx.recv_frames;
+	}
+#endif /* CONFIG_SPI_SLAVE */
+
```
Comment (tbursztyka) [inline drivers/spi/spi_dw.c:443]: Could you split this commit in 2? one for the init macro, one for the slave fixes.

Comment (XenuIsWatching) [reply drivers/spi/spi_dw.c:443]: The slave fixes are also within the init macro. It's kinda difficult to split them up

Comment (XenuIsWatching) [reply drivers/spi/spi_dw.c:443]: done

**Review by tbursztyka — DISMISSED** [review]

**Comment by XenuIsWatching** [comment]
@tbursztyka An additional issue was found, where the spi-max-xfer-size was defined by a kConfig which was looking at the ARCH of the cpu. This really came from the configuration that the IP was generated with. This adds a devicetree property for it.

**Review by cfriedt — APPROVED** [review]

**Comment by XenuIsWatching** [comment]
pinging @tbursztyka for review

**Review by carlescufi — APPROVED** [review]

**Pull request merged**